### PR TITLE
chore: Add dependabot groups

### PR DESCRIPTION
**Pull request by cmaddox5**
I want to try out Dependabot's `groups` option so we can reduce the number of PRs it opens. Feel free to call out groups that should be added or groups I added that don't look right.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -18,3 +18,35 @@ updates:
...
+          - "@babel*"
+     webpack:
+        applies-to: version-updates
+        patterns:
+          - "webpack*"
+          - "*webpack-plugin"
+     moment:
+        applies-to: version-updates
+        patterns:
+          - "moment*"
```
Comment (sloanelybutsurely) [inline .github/dependabot.yml:52]: **suggestion:**

```yaml
sentry:
  applies-to: version-updates
  patterns:
    - "@sentry/*"
```

**Review by sloanelybutsurely — APPROVED** [review]

**Comment by digitalcora** [comment]
Could we make a group that just has all patch-level updates (of all libraries) in it, using `update-types`? I feel like these both make up a majority of the "commit noise" and are a kind of update we can be reasonably confident about making "in bulk", assuming libraries are following semver conventions.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -18,3 +18,45 @@ updates:
     ignore:
       - dependency-name: "*"
         update-types: ["version-update:semver-minor"]
+    groups:
+      patches:
+        applies-to: version-updates
+        patterns:
+          - "*"
+        update-types:
+          - "patch"
```
Comment (digitalcora) [inline .github/dependabot.yml:27]: We could do this group for `mix` packages too.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -18,3 +18,45 @@ updates:
     ignore:
       - dependency-name: "*"
         update-types: ["version-update:semver-minor"]
+    groups:
+      patches:
+        applies-to: version-updates
+        patterns:
+          - "*"
```
Comment (digitalcora) [inline .github/dependabot.yml:25]: I don't think this is required, based on the examples in the docs — omitting `patterns` should be the same as having `*` as the only pattern.

**Review by digitalcora — APPROVED** [review]
Looking forward to seeing the results!

File: .github/dependabot.yml (hunk 3 above)
Comment (cmaddox5) [reply .github/dependabot.yml:25]: O gotcha. Was being a bit too cautious.

**Pull request merged**